server/server.go

```diff
@@ -495,6 +498,27 @@ func (s *Server) Start(ctx context.Context) error {
 	s.stopper.RunWorker(func() {
 		netutil.FatalIfUnexpected(m.Serve())
 	})
+
+	// We might have to sleep a bit to protect against transferring leases too
+	// soon after startup: immediately after startup, it's not OK to transfer
+	// leases away because we can't compute a proper start timestamp for the
+	// transferred lease. This timestamp needs to be higher than all the reads
+	// that have been served by this node. The problem is that, after a restart,
+	// we don't know what reads we've served before the restart - we might have
+	// served reads in the future of this node's clock because we might have been
+	// using logical timestamp coming from other nodes with a faster clock. When
+	// we restarted, our (logical) clock is reset.
+	// So, we wait up to MaxOffset. We assume we couldn't have served timestamps
+	// more than MaxOffset in the future.
+	now := timeutil.Now()
+	minStartTime := startTime.Add(s.clock.MaxOffset())
```

Comment:
I thought @tschottdorf said that we need to wait until any previously held lease has expired. I can't recall the reason for needing to do so. Did he change his mind about that?